### DISCUSSION TITLE
How to modify example transfer-06 so that it involves a file transfer.

### DISCUSSION THREAD

**Discussion by Silhim** [eclipse-edc/Samples#31] [Q&A]
Hello dear edc team,

I just completed the transfer-06 example and it worked really well. I did it over two networks and Im wondering which steps you'd have to take to make this a file transfer so that we'd have a .json file or generel file on the consumer side.
A made really naive attempt of just editing the curl request for the transfer. I adapted the "dataDestination" so that it is such as in the transfer-01 example: 
```
...
"dataDestination": { 
                    "properties": {
                        "path": "/home/operation/edcSamples/Samples/transfer/transfer-06-consumer-pull-http/http-pull-consumer",
                        "keyName": "keyName",
                        "type": "File"
                      }
...
```
But as expected this brings up an error message on the provider side: SEVERE 2023-03-15T21:10:18.212684 TransferProcessManager: Fatal error initiating data transfer: 8423eee6-0d65-452a-a952-4bb3fb12603f. Error details: Failed to delegate data transfer to Data Plane: Cannot find a transfer Service that can handle HttpData source and File destination.

Thanks in advance for the help,
best regards...

**Comment by ndr-brt** [comment]
You receive the error because the `File` type has been not registered on the data-plane instance, you'd also need a `DataSink` that supports it. If I remember correctly in the transfer-01 the whole data-plane think was jumped off with a custom data flow controller, so it can't be really applied.

**Reply by Silhim** [reply]
Hey, could you tell me what files I'd exactly have to edit? I found the HttpEndpointDataReferenceReceiver.java which is being used in transfer-06 and it uses the Mediatype Json. Would it be as easy as just chaning the Mediatype to for example "binary data"? And what are the other files which need to be changed?
Thanks a lot in advance

**Reply by ndr-brt** [reply]
You can take a look at the transfer-01 sample, in that is shown how to implement a data plane with a `File` source and sink
